### PR TITLE
initial bootstrap script for incremental compiler support

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

In order to support in browser hot reload/hot restart, we need to create a bootstrap script to invoke various dart sdk methods and attach functionality to the window. This is a first pass at what needs to be done to configure it correctly.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This isn't quite ready yet

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/web/bootstrap.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+window.\$hotReloadHook = function(modules) {
+  return new Promise(function(resolve, reject) {
+    if (modules == null) {
+      reject();
+    }
+    // If no modules change, return immediately.
+    if (modules.length == 0) {
+      resolve();
+    }
+    var reloadCount = 0;
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/web/bootstrap.dart]: but why?

[the remark this answers is not in the document]
Comment (yjbanov) [reply packages/flutter_tools/lib/src/web/bootstrap.dart]: For saner semantics? For this little amount of code it doesn't matter. LGTM either way.

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/web/bootstrap.dart]: Maybe - I"m not sure how much saner you can get in JavaScript :)

**Pull request merged**